sounddevice.py

```diff
@@ -2262,13 +2262,68 @@ def reset(self):
         vars(self).clear()
         self.__init__()
 
+
 if not hasattr(_ffi, 'I_AM_FAKE'):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
 
 
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors.
+
+    Parameters
+    ----------
+    msg : str, optional
+        User error message.
+    code : int, optional
+        PaErrorCode value or None if there is no PaErrorCode associated
+        with the error.
+
+    Notes
+    -----
+    - The `host_api`, `host_code`, and `host_text` attributes will be None
+      except for the case when `code` == `paUnanticipatedHostError`.
+    
+    Attributes
+    ----------
+    code : int or None
+        The PaErrorCode integer value or None if there is no associated error
+        code.
+    host_api : int or None
+        The index of the host API which returned the error code.
+    host_code : int or None
+        The host error code returned.
+    host_text : str or None
+        A textual description of the host error if available, otherwise a
+        zero-length string.
+    """
+    def __init__(self, msg=None, code=None):
+        self.code = code
+        self.host_code = self.host_api = self.host_text = None
+
+        if code:
+            if code == _lib.paUnanticipatedHostError:
+                info = _lib.Pa_GetLastHostErrorInfo()
+                hostapi = _lib.Pa_HostApiTypeIdToHostApiIndex(info.hostApiType)
+                apiinfo = _lib.Pa_GetHostApiInfo(hostapi)
+                self.host_code = info.errorCode
+                if apiinfo: self.host_api = _ffi.string(apiinfo.name).decode()
+                else:       self.host_api = 'unknown'
+                self.host_text = _ffi.string(info.errorText).decode()
+                errormsg = "Unanticipated '{0}' host error"\
+                  .format(self.host_api, self.host_code)
+            else:
+                errormsg = _ffi.string(_lib.Pa_GetErrorText(code)).decode()
+            errormsg = "{0} [PaErrorCode {1}]".format(errormsg, self.code)
+            if self.host_code is not None:
+                errormsg += ": '{0}' [HostErrorCode {1}]"\
+                  .format(self.host_text, self.host_code)
```

Comment:
What about using the host API name here instead of "HostErrorCode"?
Then we wouldn't need to mention the name above ...

Something like `Unanticipated host error [PaErrorCode -9999]: 'Something went wrong here' [ALSA error 795]`?